### PR TITLE
Alerting: Remove legacy alerting related provisioning

### PR DESCRIPTION
**Pull request by JacobsonMT**
**What is this feature?**

Removes provisioning of legacy alerts / notification channels and `/api/admin/provisioning/notifications/reload`.

**Why do we need this feature?**

Legacy alerting is no more.

**Special notes for your reviewer:**

Followup PRs will remove the code entirely, this is a fairly minimal changeset that focuses on removing legacy provisioning-related use.

### CODE REVIEW

File: pkg/api/admin_provisioning.go [hunk 1]
```diff
@@ -94,6 +94,9 @@ func (hs *HTTPServer) AdminProvisioningReloadPlugins(c *contextmodel.ReqContext)
 // 403: forbiddenError
 // 500: internalServerError
 func (hs *HTTPServer) AdminProvisioningReloadNotifications(c *contextmodel.ReqContext) response.Response {
+	if !hs.Cfg.IsLegacyAlertingEnabled() {
+		return response.Error(http.StatusNotFound, "Legacy alerting is not enabled", nil)
```
Comment (rwwiv) [inline pkg/api/admin_provisioning.go:98]: Nit: I wonder if there's a more appropriate status code for this.

**Review by yuri-tceretian — APPROVED** [review]
LGTM

File: pkg/services/dashboards/service/dashboard_service.go [hunk 2]
```diff
@@ -319,7 +319,7 @@ func (dr *DashboardServiceImpl) SaveProvisionedDashboard(ctx context.Context, dt
 
 	dto.User = accesscontrol.BackgroundUser("dashboard_provisioning", dto.OrgID, org.RoleAdmin, provisionerPermissions)
 
-	cmd, err := dr.BuildSaveDashboardCommand(ctx, dto, dr.cfg.IsLegacyAlertingEnabled(), false)
+	cmd, err := dr.BuildSaveDashboardCommand(ctx, dto, false, false)
```
Comment (yuri-tceretian) [inline pkg/services/dashboards/service/dashboard_service.go:322]: as long as we delete the flag, probably it makes sense to either leave a \\TODO or refactor the method and remove the flag at all?

File: pkg/api/api.go [hunk 3]
```diff
@@ -596,7 +596,6 @@ func (hs *HTTPServer) registerRoutes() {
 		adminRoute.Post("/provisioning/dashboards/reload", authorize(ac.EvalPermission(ActionProvisioningReload, ScopeProvisionersDashboards)), routing.Wrap(hs.AdminProvisioningReloadDashboards))
 		adminRoute.Post("/provisioning/plugins/reload", authorize(ac.EvalPermission(ActionProvisioningReload, ScopeProvisionersPlugins)), routing.Wrap(hs.AdminProvisioningReloadPlugins))
 		adminRoute.Post("/provisioning/datasources/reload", authorize(ac.EvalPermission(ActionProvisioningReload, ScopeProvisionersDatasources)), routing.Wrap(hs.AdminProvisioningReloadDatasources))
-		adminRoute.Post("/provisioning/notifications/reload", authorize(ac.EvalPermission(ActionProvisioningReload, ScopeProvisionersNotifications)), routing.Wrap(hs.AdminProvisioningReloadNotifications))
```
Comment (yuri-tceretian) [inline pkg/api/api.go:599]: Please remove `ScopeProvisionersNotifications` as it is not used anywhere

File: pkg/services/dashboards/service/dashboard_service.go (hunk 2 above)
Comment (yuri-tceretian) [reply pkg/services/dashboards/service/dashboard_service.go:322]: We will remove it in a follow up

**Comment by yuri-tceretian** [comment]
NIT: there is mock https://github.com/grafana/grafana/blob/025b2f3011821bb7e6ab4781e252f3a1ecd19130/pkg/services/provisioning/provisioning_mock.go#L59 that we can update and remove unused method

**Review by yuri-tceretian — APPROVED** [review]
LGTM

File: pkg/api/api.go (hunk 3 above)
Comment (JacobsonMT) [reply pkg/api/api.go:599]: There's a decent amount of unused code now after this PR, removing it all affects a few dozen files though. I'll do it in a followup PR.

**Comment by JacobsonMT** [comment]
> NIT: there is mock
> 
> https://github.com/grafana/grafana/blob/025b2f3011821bb7e6ab4781e252f3a1ecd19130/pkg/services/provisioning/provisioning_mock.go#L59
> that we can update and remove unused method

Had to add the interface back to get enterprise tests passing. Will remove it with the rest of the unused code after removing the dependency in enterprise.

**Pull request merged**